src/keys/msgsig.rs

```diff
@@ -0,0 +1,128 @@
+
+/// Signing arbitrary messages and verify message signatures
+
+use bitcoin::hashes::Hash;
+use bitcoin::secp256k1::recovery::RecoverableSignature;
+use bitcoin::secp256k1::{All, Message, Secp256k1, SecretKey};
+use bitcoin::util::misc::{signed_msg_hash, MessageSignature};
+use bitcoin::{Address, Network, PrivateKey, PublicKey};
+
+/// Trait for message signers
+pub trait MessageSigner<S> {
+    /// Sign an arbitrary message
+    fn sign(&self, msg: &str) -> S;
+}
+
+/// Trait for message signature verifiers
+pub trait MessageSignatureVerifier<S> {
+    /// Verify a signature against an arbitrary message
+    fn verify(&self, sig: S, msg: &str) -> bool;
+}
+
+/// A message signer using ECDSA
+pub struct EcdsaMessageSigner {
+    secp: Secp256k1<All>,
+    secret_key: SecretKey,
+}
+
+impl EcdsaMessageSigner {
+
+    /// Creates message signer from a bitcoin ECDSA private key
+    pub fn from_prv(prv: PrivateKey) -> Self {
+        Self::from_secret_key(prv.key)
+    }
+
+    /// Creates message signer from an ECDSA secret key
+    pub fn from_secret_key(secret_key: SecretKey) -> Self {
+        EcdsaMessageSigner {
+            secret_key,
+            secp: Secp256k1::new(),
+        }
+    }
+}
+
+impl MessageSigner<RecoverableSignature> for EcdsaMessageSigner {
+    fn sign(&self, message: &str) -> RecoverableSignature {
+        let msg_hash = signed_msg_hash(message);
+        self.secp.sign_recoverable(
+            &Message::from_slice(&msg_hash.into_inner()[..]).unwrap(),
+            &self.secret_key,
+        )
+    }
+}
+
+/// A message signature verifier using ECDSA
+pub struct EcdsaMessageSignatureVerifier {
+    secp: Secp256k1<All>,
```

Comment:
I would just take a reference here to avoid cloning the context. If you don't want to add a lifetime to the struct you could just change the api of `verify()` to take a secp argument

Reply:
I mean, change the member declaration to be a reference and then take one in `from_pub` and `from_address`

Reply:
I don't understand what do you mean here, can you elaborate a bit more?

My interpretation is that`verify()` doesn't have to be a method on the wallet, because you don't need anything from the wallet to verify a message (while for signing you need the private keys)

Reply:
I intially wanted to have similar interfaces for signing and verification. Therfore, I thought if signing is moved to the wallet, so should also verification. Combined with my assumption of having schnorr for message signatures at some point, BDK would end up with multiple `verify` methods.

However, this is not the case since schnorr is aparently not planned. Moreover, you  are right that `verify` does not have to be on the wallet - except you want verification to happen with the pubkey of a different `pk(...)` wallet. But this feels a little odd.

Therefore, I suggest that we change the PR to have 
* a method on the wallet for message signing which uses all private keys of the wallet
* message signature verification which is decoupled from the wallet